### PR TITLE
[RFC, do not merge] run black on core/audio.py

### PR DESCRIPTION
**Pull request by lostanlen**
#### Reference Issue
Related to #1070 

#### What does this implement/fix? Explain your changes.
This PR is the raw result of running `black librosa/core/audio.py` on the master branch.
It's meant to serve as a conversation starter around #1070
We can configure black a little bit more finely in the future depending on what we like.

### Any other comments?
* i like the new formatting of the function prototype, with every argument on a line of its own.

* double quotes vs single quotes: should we let black standardize this? right now it puts everything to double quotes.

* line 215 is a bit weird: `frame = frame[: s_end - n_prev]`. do we really need space after this colon?

* note that black adds whitespace around _every_ operator. Yet I tend to remove the whitespace to signify operator precedence. So black would destroy that visual cue.
`den = q*den - bwd_pred_error[-1]**2 - fwd_pred_error[0]**2`
becomes `den = q * den - bwd_pred_error[-1] ** 2 - fwd_pred_error[0] ** 2`

* compound return statements become parenthesized and multiline.
Before:
```python
    return np.pad((y_sign[tuple(slice_post)] != y_sign[tuple(slice_pre)]),
                  padding,
                  mode='constant',
                  constant_values=pad)
```

After:
```python
    return np.pad(
        (y_sign[tuple(slice_post)] != y_sign[tuple(slice_pre)]),
        padding,
        mode="constant",
        constant_values=pad,
    )
```

Before:
```python
        return np.digitize(x_comp,
                           np.linspace(-1, 1, num=int(1+mu), endpoint=True),
                           right=True) - int(mu + 1)//2
```
After:
```python
        return (
            np.digitize(
                x_comp, np.linspace(-1, 1, num=int(1 + mu), endpoint=True), right=True
            )
            - int(mu + 1) // 2
        )
```
It almost makes me think that we want to assign a variable before applying the `return` statement.

* On this example (`core/audio.py`), the overall line length remained almost the same. 1521 lines before and 1574 lines after. But the number of additions/deletions is quite high, mostly because of single quote to double quote conversion.

* Worth watching: https://www.youtube.com/watch?v=esZLCuWs_2Y

* Thoughts?

### CODE REVIEW

File: librosa/core/audio.py [hunk 1]
```diff
@@ -345,20 +370,20 @@ def stream(path, block_length, frame_length, hop_length,
...
 
     if not (np.issubdtype(type(block_length), np.integer) and block_length > 0):
-        raise ParameterError('block_length={} must be a positive integer')
+        raise ParameterError("block_length={} must be a positive integer")
     if not (np.issubdtype(type(frame_length), np.integer) and frame_length > 0):
-        raise ParameterError('frame_length={} must be a positive integer')
+        raise ParameterError("frame_length={} must be a positive integer")
     if not (np.issubdtype(type(hop_length), np.integer) and hop_length > 0):
-        raise ParameterError('hop_length={} must be a positive integer')
+        raise ParameterError("hop_length={} must be a positive integer")
```
Comment (bmcfee) [inline librosa/core/audio.py]: whoa, these raise lines are wrong!  there's no `.format(hop_length)` (etc) to interpolate the {} to actual values.

This is independent of the issue for this PR, but good to fix anyway.

**Comment by bmcfee** [comment]
>     * i like the new formatting of the function prototype, with every argument on a line of its own.

I agree in general, but it does have the unfortunate side effect of breaking folds in vim:

image

This is probably something to be fixed on my side, but I wanted to point it out.


>     * double quotes vs single quotes: should we let black standardize this? right now it puts everything to double quotes.

I generally prefer single, but I'm okay with double.  It's mostly a question of which one requires more escaping (ie due to nested punctuation).

>     * line 215 is a bit weird: `frame = frame[: s_end - n_prev]`. do we really need space after this colon?

I don't think that we do, but it doesn't hurt.



>     * note that black adds whitespace around _every_ operator. Yet I tend to remove the whitespace to signify operator precedence. So black would destroy that visual cue.

I prefer your way, though it can get messy when compounded.

> It almost makes me think that we want to assign a variable before applying the `return` statement.

This is probably a good thing to do anyway.

File: librosa/core/audio.py (hunk 1 above)
Comment (lostanlen) [reply librosa/core/audio.py]: good catch ...

**Comment by bmcfee** [comment]
#1193 implements code styling, so I think we can close this PR out.  (It seemed easier to do it that way than to rebase this one.)

**Pull request closed without merging**